### DISCUSSION TITLE
3 party cookie blocked via running cli tool

### DISCUSSION THREAD

**Discussion by Joeyindigital** [GoogleChromeLabs/ps-analysis-tool#610] [Support Forum] [answered]
**Describe the bug**
While running the single URL (I use google.com here) scan using PSAT cli tool, the result comes out differently than the psat extension. There are no third-party cookies exist in the dashboard. I do run the tool in Windows 11 with Windows power shell. 

Here are the screenshots of my code.
![image](https://github.com/GoogleChromeLabs/ps-analysis-tool/assets/166688925/9d45a356-5647-435d-a59d-39868952ee38)
![image](https://github.com/GoogleChromeLabs/ps-analysis-tool/assets/166688925/d1a4e92f-36aa-40c8-9724-28224120ed54)
![image](https://github.com/GoogleChromeLabs/ps-analysis-tool/assets/166688925/1f308934-cfe4-48ce-828b-3ebd7aa3d5af)

**Comment by milindmore22** [comment]
Hello @Joeyindigital

This likely means google.com doesn't use them, so the PSAT Chrome extension isn't finding any.  However, there can be slight differences between the PSAT extension and the command-line tool (CLI).

Several factors can affect how cookies appear, like the testing environment, analysis duration, and user actions on the page.

For more on PSAT CLI vs extension discrepancies, see our [wiki](https://github.com/GoogleChromeLabs/ps-analysis-tool/wiki/PSAT-Command-Line-Interface#discrepancy-between-cli-and-extension).

**Comment by Joeyindigital** [comment]
Hey Milind,However I could see that if I run the cli tool with different url it also appears to not show the 3rd party cookies. Any suggestions to pass this issue?JoeySent from my iPhoneOn 11 Apr 2024, at 9:21 PM, Milind More ***@***.***> wrote:﻿
Hello @Joeyindigital
This likely means google.com doesn't use them, so the PSAT Chrome extension isn't finding any.  However, there can be slight differences between the PSAT extension and the command-line tool (CLI).
Several factors can affect how cookies appear, like the testing environment, analysis duration, and user actions on the page.
For more on PSAT CLI vs extension discrepancies, see our wiki.

**Comment by milindmore22** [comment]
Hey @Joeyindigital, that's definitely odd! 

Let's make sure that you are using the latest version, please switch to the PSAT directory (cloned repo) and run the command `git pull` in your terminal.

After the `git pull` is completed, execute `npm i && npm run cli:build` this will re-install or update dependencies and rebuild your CLI application, once the build is completed check by running `npm run cli -- -u https://bbc.com -nt`

If you're testing specific URLs, please provide them. This allows us to attempt replicating the issue.

**Comment by Joeyindigital** [comment]
Hi @milindmore22 The PSAT is the latest version @0.6.0. I run the comment npm run cli -- -u https://bbc.com -nt and the dashboard comes back with no data.
![image](https://github.com/GoogleChromeLabs/ps-analysis-tool/assets/166688925/24156560-a69e-440d-bf9b-0fd9b17b0861)
![image](https://github.com/GoogleChromeLabs/ps-analysis-tool/assets/166688925/480c7dfc-5359-4398-a31c-bd5972283ec0)

**Comment by milindmore22** [comment, marked as answer]
Hello @Joeyindigital 

Thank you for sharing the screenshots, it appears that you are using a downloaded PSAT repo, as git was unable to fetch the latest changes. 

Can you please make sure to clone the repo by running the following command.
1. `git clone https://github.com/GoogleChromeLabs/ps-analysis-tool.git` 
2. `cd ps-analysis-tool`
3. `npm i`
4. `npm run cli:build`

For details installation instructions, please check our [Wiki](https://github.com/GoogleChromeLabs/ps-analysis-tool/wiki/PSAT-Command-Line-Interface#getting-started-with-psat-cli)

I will be converting this issue to support forum discussion, For future reference, please [create support discussion](https://github.com/GoogleChromeLabs/ps-analysis-tool/discussions/new?category=support-forum) if you have any question about the PSAT.
